docs/proposals/proposal-bug-1512042

```diff
@@ -0,0 +1,80 @@
+
+# Problem Description
+The problem is that we want to be able to control how the privileges get escalated into that namespace to see/pull the images. This means that we will need a robust solution, that is not a simple bug fix.
+
+## Implementation Details
+The proposed changes are adding a configuration option to the local_registry. This configuration option will be similar to the auto_escalate feature we already have. The option will allow for checking if the originating user has access to the customer namespace to pull images. This will be the default, and you can specify to auto grant access to the namespace. 
+
+To grant access to sandbox's service account we will grant access to pull images from the custom namespace. 
+
+There is also a third option that we should document, which is to allow everyone to access your images in that custom namespace. This option will work even with the default because the user will always have access then.
+
+### Open Questions
+1. I think this check should be in the handler. This is an authorization check in my mind and should be treated as such.
+2. The runtime package should be used to determine the check, something like `CheckAccessToImage`.
+3. We need to figure out how to handle if the adapter is any other adapter then local_openshift. 
+    1. Add a field on the spec to state the registry/adapter that it is using?
+    2. We will also need to track the namespace that it should check.
+    3. Need to be able to not run this check if the not available.
+    4. Need this to be usable by other adapters if they want (ecr adapter could use this to look at IAM access of the originating origin user, gcr adapter could use their Cloud IAM). Or a calller could want to do their own authorization check using our built in auth model (basic auth example).
+
+### Possible Solutions
+1. Create new fields on the spec. These fields will keep track of the adapters/registries that added the spec. Add a function in the runtime package that will be called by the handler for authorization. This function will take the spec, the originating user structure, and some undefined structures as a variadic. Example:
+```go
+type RuntimeAuthorization func(apb.Spec, apb.UserInfo, interface{}...)
+
+var RuntimeAuthorization RuntimeAuthorization
+```
+2.  Create new fields on the spec sor that it can track the adapters/registries that added this specific spec. Add a interface Authorization for the runtime package. This will have a single method, Authorize that will take a UserInfo struct. This then can be chained, and the runtime package should manage this as this is supposed to be exposed so everyone can add one. We can helper methods to add providers to the authorization on the fly.
```

Comment:
`sor -> so`

Reply:
Think the name of `Authorization` needs to be more specific if it's going to live in the runtime package. `runtime.Authorization` could mean anything.

Reply:
`can helper -> can have helper`

Reply:
+1 to the chain idea.

Reply:
nit: Add a interface Authorization --> Add an interface

Reply:
@eriknelson can you explain a little more to me why `runtime.Authorization` does not work. In my mind, we are saying that the only authorization that will be done is by the managed authorization checks at runtime. This means that the handler calls a single function and all the authorization happens in that chain. What other authorizations could be happing and what ideas do you have to make it more specific?